File: Mandelbrot.py
import pygame as pg
import math

def Mandelbrot(x, y, x0, y0, n_max):
    Re = x
    Im = y
    Re_temp = 0
    Im_temp = 0

    n = 0
    while n < n_max and Re**2 + Im**2 < 4:
        Re_temp = Re**2 - Im**2 + x0
        Im = 2*Re*Im + y0
        Re = Re_temp
        n += 1
    return colorgen(n, n_max)

def draw(screen, x, y, color, w, h):
    #screen.set_at((x, y), color)
    if(x > w or y > h):
        return
    pg.draw.rect(screen, color, pg.Rect(x-1, y-1, 1, 1))
def colorgen(n, n_max):
    red = round(255*n/n_max)
    green = round(255*n/n_max)
    blue = 0
    return (red, green, blue)

def main():
    (width, height) = (800, 600)
    pg.display.set_caption('Mandelbrot')
    screen = pg.display.set_mode((0,0), pg.FULLSCREEN | pg.DOUBLEBUF)
    dim = pg.display.Info()
    screen.set_alpha(None)
    w = dim.current_w*2
    h = dim.current_h*2
    for x in range(w+1):
        for y in range(round(h/2)+1):
            color = Mandelbrot(0, 0, 3*x/w-2, 2*y/h-1, 50)
            draw(screen, x, y, color, w, h)
            draw(screen, x, (h-y), color, w, h)

    running = True
    while running:
        running = not user_quit()
        pg.display.flip()

def user_quit():
    for event in pg.event.get():
        if event.type == pg.QUIT:
            return True
        elif event.type == pg.KEYDOWN:
            if event.key == pg.K_q:
                return True
    return False



if __name__ == '__main__':
    main()

File: mandelbrot.js
var canvas = document.querySelector('canvas');
var c = canvas.getContext('2d');

canvas.width = window.innerWidth;
canvas.height = window.innerHeight;
console.log(window.innerWidth + ":" + window.innerHeight);
function mandelbrot(x, y, a, b){
  var Re = x;
  var Im = y;
  var n = 0;
  var Re_temp;
  var Im_temp;
  while(n <= 100 && (Re*Re + Im*Im < 4)){
    Re_temp = Re*Re - Im*Im + a;
    Im = 2*Re*Im + b;
    Re = Re_temp;
    n++
  }
  return n;
}

function outsidepixel(x, y, n){
  if(n >= 100){
    n = 0
  }
  var color = 'rgb('+ Math.floor(255*n/100) +', ' + Math.floor(100*n/100) + ', 0)';
  c.beginPath();
  c.rect(x, y, 1, 1);
  c.rect(x, window.innerHeight-y, 1, 1);
  c.fillStyle = color;
  c.fill();
  c.closePath();
}

for(var x = 0; x <= window.innerWidth; x+=1){
  for(var y = 0; y <= window.innerHeight/2; y+=1){
    //console.log(mandelbrot(0, 0, 3*x/window.innerWidth-2, 2*y/window.innerHeight-1));
    outsidepixel(x, y, mandelbrot(0, 0, 3*x/window.innerWidth-2, 2*y/window.innerHeight-1));
  }
}
